cogs/reward.py

```diff
@@ -11,66 +11,123 @@ def __init__ (self, bot):
     @commands.command()
 
     async def reward(self,ctx, timeString=None, tier=None):
-        rewardCommand = f"\nPlease follow this format:\n```yaml\n{commandPrefix}reward \"#h#m\" \"tier\"```\n"
+        rewardCommand = f"\nPlease follow this format:\n```yaml\n{commandPrefix}reward \"#h#m\" \"tier or character name\"```\n"
 
         def convert_to_seconds(s):
             return int(s[:-1]) * seconds_per_unit[s[-1]]
 
         channel = ctx.channel
+        author = ctx.author
+        charEmbed = discord.Embed()
+        charEmbedmsg = None
+        char = tier #this is mainly for my sanity
+        
         if timeString is None:
             await channel.send(content="Woops, you're forgetting the time duration for the command." + rewardCommand)
             return
 
         if tier is None:
-            await channel.send(content="Woops, you're forgetting the tier for the command. Please try again with 1, 2, 3, or 4 or Junior, Journey, Elite, or True as the tier, and" + rewardCommand)
+            await channel.send(content="Woops, you're forgetting the tier or character name for the command. Please try again with 1, 2, 3, or 4 or Junior, Journey, Elite, or True as the tier, or use a character name and" + rewardCommand)
             return
 
         seconds_per_unit = { "m": 60, "h": 3600 }
         lowerTimeString = timeString.lower()
+        
+        characterPresent = None
 
         if tier not in ('0','1','2','3','4', '5') and tier.lower() not in [r.lower() for r in roleArray]:
-            await channel.send(f"**{tier}** is not a valid tier. Please try again with **New** or **0**, **Junior** or **1**, **Journey** or **2**, **Elite** or **3**, **True** or **4**, or **Ascended** or **5**.")
-            return
-
-        tierName = ""
-        if tier.isdigit():
-            tierName = roleArray[int(tier)]
-            tier = tierName
-        else:
-            tierName = tier.capitalize()
-        print(tierName)
-        l = list((re.findall('.*?[hm]', lowerTimeString)))
-        totalTime = 0
-        for timeItem in l:
-            totalTime += convert_to_seconds(timeItem)
+            charDict, charEmbedmsg = await checkForChar(ctx, char, charEmbed)
+            if charDict == None:
+                await channel.send(f"**{tier}** is not a valid tier or character name. Please try again with **New** or **0**, **Junior** or **1**, **Journey** or **2**, **Elite** or **3**, **True** or **4**, or **Ascended** or **5**, or input a valid character name.")
+                return
+            else:
+                characterPresent = True
+        
+        if characterPresent == None:
```

Comment:
Combined with the change to `characterPresent` above this should be reduced to just 
`if not characterPresent:`
and the `elif` should be an else